barretenberg/cpp/src/barretenberg/vm2/simulation/lib/raw_data_dbs.hpp

```diff
@@ -99,6 +100,76 @@ class HintedRawMerkleDB final : public LowLevelMerkleDBInterface {
     AppendLeafResult appendLeafInternal(MerkleTreeId tree_id, const FF& leaf);
 };
 
+class PureRawMerkleDB : public LowLevelMerkleDBInterface {
+  public:
+    PureRawMerkleDB(world_state::WorldStateRevision ws_revision, TreeSnapshots tree_snaphots)
+        : tree_roots(tree_snaphots)
+        , ws_revision(ws_revision)
+    {}
+
+    // Helper to build the PureRawMerkleDB from the WorldState, with support for error handling.
+    static std::optional<PureRawMerkleDB> connect(world_state::WorldStateRevision ws_revision)
+    {
+        if (!world_state::WorldStateManager::has_world_state()) {
+            return std::nullopt;
+        }
+        // Since we checked above, it is safe to get the instance.
+        world_state::WorldState* ws_instance = world_state::WorldStateManager::get_world_state();
+        auto nullifier_tree = ws_instance->get_tree_info(ws_revision, MerkleTreeId::NULLIFIER_TREE);
+        auto note_hash_tree = ws_instance->get_tree_info(ws_revision, MerkleTreeId::NOTE_HASH_TREE);
+        auto public_data_tree = ws_instance->get_tree_info(ws_revision, MerkleTreeId::PUBLIC_DATA_TREE);
+        auto l1_to_l2_message_tree = ws_instance->get_tree_info(ws_revision, MerkleTreeId::L1_TO_L2_MESSAGE_TREE);
+
+        // Initialize the TreeSnapshots
+        auto tree_roots = TreeSnapshots {
```

Comment:
AFAICT these roots are
1. not kept up to date
2. why does this factory do this and not the constructor?

I think I would not track the roots, and make `get_tree_roots` actually execute the above code.

If you need to remove the & from the result in `get_tree_roots`, go ahead.

Reply:
They were being updated as part of the appends/inserts. I don't mind switching it to the `get_tree_roots` approach (that was my preferred if not for the `&` return for that function).

The factory was just so that I could handle errors in the simulation_helper, throwing in the constructor felt bad. I think much of this will change with the new WorldState approach so i can get rid of this too